packages/lesswrong/lib/collections/dialogueChecks/collection.ts

```diff
@@ -0,0 +1,36 @@
+import {ensureIndex} from "../../collectionIndexUtils";
+import {addUniversalFields, getDefaultMutations, getDefaultResolvers, schemaDefaultValue} from "../../collectionUtils";
+import {createCollection} from "../../vulcan-lib";
+import schema from "./schema";
+
+export const DialogueChecks: DialogueChecksCollection = createCollection({
+  collectionName: 'DialogueChecks',
+  typeName: 'DialogueCheck',
+  collectionType: 'pg',
+  schema,
+  resolvers: getDefaultResolvers('DialogueChecks'),
+  logChanges: true,
+})
+
+DialogueChecks.checkAccess = async (user: DbUser|null, document: DbDialogueCheck, context: ResolverContext|null): Promise<boolean> => {
+  // Case 1: A user can see their own checks
+  if (document.userId === user?._id) {
+    return true;
+  }
+
+  // Case 2: A user can see the checks of people they themselves have checked... 
+  const existingCheck = await DialogueChecks.findOne({ userId: user?._id, targetUserId: document.userId, checked: true });
```

Comment:
I might call this `reciprocalCheck` or something to make it more clear what you're trying to look up